### DISCUSSION TITLE
What is the best approach for upgrading on VSC?

### DISCUSSION THREAD

**Discussion by mrjeth** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#490] [Q&A]
The board seem to work! I have recently compiled and briefly verified that it works! I am super excited to test it with water and heat! Since it is still somewhat cold outside, I will wait another couple of weeks before filling the SPA with Water. 

The board I am using is the Wemos d1 mini without the PCB the cables are connected directly. 
I can hear the short intro song. There is information on the screen and I can touch the buttons (but they are very sensitive) 
However, I have now discovered that there is a newer version of the software and before I finally close the lid to the pump I would like to have the latest Software installed.

I am a novice when it comes to VSC and Platform IO and here is my question: 
What is the best approach to handle the source code when upgrading?
 I know it is possible to upgrade using OTA but it is still the same problem. If I also understand it will be possible to upgrade using the gui, but only if the data ports are used in the same order as the original.

1)	Rename the old files and unzip the new files in parallel. Then try to find what changes you have made and change the files. Compile and upload
2)	Rename the old files and unzip the new files in parallel. Copy the changed files and replace the old . Compile and upload
3)	Some smart way that I don’t know 
4)	Other
I assume it is 1) 

Best regards 
Tomas

**Comment by mrjeth** [comment]
I looked more carefully in the new instructions and it seems that all configuration will be done through the GUI and not in the source code.

**Reply by visualapproach** [reply]
D1mini is pin compatible with nodeMCU so you can update via web UI. 
To answer your question - you do exactly as you did the first time. Download project from github, open in platformio and build code and filesystem and upload. There is no "update magic", you just upload the new firmware and filesystem as the first time.

**Reply by visualapproach** [reply]
If you use the web UI to update firmware, all settings will be untouched. So it's the preferred method

**Comment by mrjeth** [comment]
Most appreciated!
Thanks also for the superior project!

Den tis 11 apr. 2023 18:09visualapproach ***@***.***> skrev:
